pwnagotchi/plugins/default/AircrackOnly.py

```diff
@@ -1,57 +1,33 @@
-__author__ = 'pwnagotchi [at] rossmarks [dot] uk'
-__version__ = '1.0.0'
-__name__ = 'AircrackOnly'
-__license__ = 'GPL3'
-__description__ = 'confirm pcap contains handshake/PMKID or delete it'
-
 '''
 Aircrack-ng needed, to install:
 > apt-get install aircrack-ng
 '''
-
 import logging
 import subprocess
-import string
-import re
 import os
 
+__author__ = 'pwnagotchi [at] rossmarks [dot] uk'
+__version__ = '1.0.0'
+__name__ = 'AircrackOnly'
+__license__ = 'GPL3'
+__description__ = 'confirm pcap contains handshake/PMKID or delete it'
+
 OPTIONS = dict()
 
+
 def on_loaded():
     logging.info("aircrackonly plugin loaded")
 
 def on_handshake(agent, filename, access_point, client_station):
-    display = agent._view
-    todelete = 0
-
-    result = subprocess.run(('/usr/bin/aircrack-ng '+ filename +' | grep "1 handshake" | awk \'{print $2}\''),shell=True, stdout=subprocess.PIPE)
-    result = result.stdout.decode('utf-8').translate({ord(c) :None for c in string.whitespace})
-    if result:
-        logging.info("[AircrackOnly] contains handshake")
-    else:
-        todetele = 1
-
-    if todelete == 0:
-        result = subprocess.run(('/usr/bin/aircrack-ng '+ filename +' | grep "PMKID" | awk \'{print $2}\''),shell=True, stdout=subprocess.PIPE)
-        result = result.stdout.decode('utf-8').translate({ord(c) :None for c in string.whitespace})
-        if result:
-            logging.info("[AircrackOnly] contains PMKID")
-        else:
-            todetele = 1
-
-    if todelete == 1:
-        os.remove(filename)
-        set_text("uncrackable pcap")
-        display.update(force=True)
-
-text_to_set = "";
-def set_text(text):
-    global text_to_set
-    text_to_set = text
-
-def on_ui_update(ui):
-    global text_to_set
-    if text_to_set:
+    ui = agent.view()
+    process = subprocess.Popen(f"/usr/bin/aircrack-ng {filename} | grep -E ([^0]+ handshake|contains handshake)", shell=True, stdin=None,
```

Comment:
Should probably be `grep -E ([^0]+ handshake|contains handshake|PMKID)` to also include pcaps containing PMKIDs.

Reply:
The "contains handshake" should catch them

Reply:
Just checked, the PMKID match looks like this
 `......... WPA (1 handshake, with PMKID)`
So the `[^0] handshake` should be enough.

Reply:
`0 handshake, with PMKID` is also possible

Reply:
ahhhh, right >.<

Reply:
thx mate!

Reply:
You're welcome! Also I've just noticed that `[^0]+ handshake` won't match `10 handshake`, which is a super improbable edge case so maybe it's not even worth fixing, but just in case, `[1-9]\d* handshake` or `[^0]\d* handshake` should work better. Totally your call though :)

Reply:
nice catch :D